server/zoom/www/cache/application_state_cache.py

```diff
@@ -62,14 +62,54 @@ def _load(self):
             self._cache.application_states)
         self._cache.remove_deletes()
 
+    def _build_default_override_store(self, override_nodename):
+        logging.debug('Override storage node not found, creating')
+        _template = {}
+        _template['grayed'] = {}
+        _template['pd_disabled'] = {}
+        self._zoo_keeper.create(override_nodename, json.dumps(_template), makepath=True)
+
+    def _update_override_info(self, path, override_key, override_value):
+        """
+        Update zookeeper with override values passed in
+        """
+        logging.debug("_update_override_info: path: {0}, override_key: {1} override_value: {2}"
+                      .format(path, override_key, override_value))
+        try:
+            _state, _unused = self._zoo_keeper.get(self._configuration.override_node)
+            if _state:
+                try:
+                    # grab current state object
+                    _current_state = json.loads(_state)
+                    _d = {}
+                    if override_value:
+                        _d[path] = override_value
+                    else:
+                        _current_state[override_key].pop(path)
+                    _current_state[override_key].update(_d)
+                    self._zoo_keeper.set(self._configuration.override_node,
+                                        json.dumps(_current_state))
+                except (TypeError, ValueError) as err:
+                    logging.critical(err)
+            else:
+                pass
+        except NoNodeError as err:
+            logging.debug('Unable to find ')
+            self._build_default_override_store(self._configuration.override_node)
+            self._update_override_info(path, override_key, override_value)
+
     def manual_update(self, path, key, value):
         """
         Manual override from client of specific value
         :type path: str
         :param key: str
         """
+        # Set the permanent storage
+
         state = self._cache._application_states.get(path, None)
         if state is not None:
+            if key in ['grayed', 'pd_disabled']:
```

Comment:
Do we have to hard-code this? If we add some new value we'll have to touch this later. Are we positive we only want to persist 'grated' and 'pd_disabled'?